source/opt/inst_bindless_check_pass.cpp

```diff
@@ -384,6 +426,29 @@ uint32_t InstBindlessCheckPass::GenLastByteIdx(ref_analysis* ref,
             });
         USE_ASSERT(found && "member offset not found");
         curr_offset_id = builder->GetUintConstantId(member_offset);
+        // Look for matrix stride for this member if there is one. The matrix
+        // stride is not on the matrix type, but in a OpMemberDecorate on the
+        // enclosing struct type at the member index. If none found, reset
+        // stride to 0.
+        found = !get_decoration_mgr()->WhileEachDecoration(
```

Comment:
I think I would rather revert the sign of the return values inside the body of the lambda.  It would improve readability of this predicate.

Reply:
Agreed, although inverting and reinverting the booleans won't quite work because WhileEachDecoration behaves differently depending on the booleans returned. So I created a new function FindDecoration which uses WhileEachDecoration as a base but inverts it, giving us just the behavior we want, which is to terminate when the decoration we are seeking is found ie when true is returned. I think you will see the readability you were looking for.

Reply:
> Agreed, although inverting and reinverting the booleans won't quite work because WhileEachDecoration behaves differently depending on the booleans returned. So I created a new function FindDecoration which uses WhileEachDecoration as a base but inverts it, giving us just the behavior we want, which is to terminate when the decoration we are seeking is found ie when true is returned. I think you will see the readability you were looking for.

Oh, right.  The early exit on `false`, right?  The new variant is indeed more readable.  Thanks.